File: langchain-python/09-multi-agent/agents/reviewer_agent.py
"""审查 Agent - 代码审查"""

from .base_agent import BaseAgent, AgentMessage


class ReviewerAgent(BaseAgent):
    """审查 Agent，擅长代码审查"""
    
    def __init__(self, llm):
        super().__init__("Reviewer", "专业代码审查员", llm, [])
    
    async def process_message(self, message: AgentMessage) -> AgentMessage:
        """处理消息"""
        code_content = message.content
        
        print(f"\n[{self.name}] 接到审查任务")
        
        review_prompt = f"""你是一个专业的代码审查员，擅长：
1. 检查代码质量
2. 识别潜在问题
3. 提供改进建议
4. 评估代码性能

代码内容：
{code_content}

请提供详细的审查报告，包括：
1. 代码质量评估
2. 发现的问题
3. 改进建议
4. 性能优化建议"""
        
        response = await self.llm.ainvoke(review_prompt)
        review_report = response.content
        
        print(f"[{self.name}] 审查完成")
        print(f"\n{'─'*60}")
        print(f"🔍 [{self.name}] 审查报告：")
        print("─"*60)
        print(review_report[:500] + "..." if len(review_report) > 500 else review_report)
        print("─"*60)
        print("\n"*3)
        
        return self.send_message(
            message.sender,
            review_report
        )

File: langchain-typescript/src/09-multi-agent/reviewer-agent.ts
import { ChatOpenAI } from "@langchain/openai";
import { BaseAgent } from "./base-agent";
import { AgentMessage } from "./types";

export class ReviewerAgent extends BaseAgent {
  constructor(llm: ChatOpenAI) {
    super("Reviewer", "代码审查和质量检查专家", llm);
  }

  async processMessage(message: AgentMessage): Promise<AgentMessage | null> {
    const codeContent = message.content;

    console.log(`\n[${this.name}] 接到任务：审查代码`);

    const reviewPrompt = `你是一个专业的代码审查员，擅长：
1. 检查代码质量
2. 识别潜在问题
3. 提供改进建议
4. 评估代码性能

请审查以下内容：
${codeContent}

审查要点：
1. 代码正确性
2. 代码风格
3. 性能优化
4. 错误处理
5. 最佳实践

请提供详细的审查报告和改进建议。`;

    const response = await this.llm.invoke(reviewPrompt);
    const reviewReport = response.content as string;

    console.log(`[${this.name}] 审查完成`);
    console.log(`\n${"─".repeat(60)}`);
    console.log(`🔍 [${this.name}] 审查报告：`);
    console.log("─".repeat(60));
    console.log(reviewReport);
    console.log("─".repeat(60));
    console.log("\n".repeat(3));

    return this.sendMessage(
      message.sender,
      `审查报告：\n${reviewReport}`
    );
  }
}
